`panic: Unhandled case in Node.Text: *ast.ComputedPropertyName` when enable `typescript/no-deprecated`

**Command (with enabled `typescript/no-deprecated` in config):**
```bash
> oxlint . --type-aware --max-warnings=0
```
**Error:**
<details>
 <summary>
panic: Unhandled case in Node.Text: *ast.ComputedPropertyName
 </summary>

```
panic: Unhandled case in Node.Text: *ast.ComputedPropertyName

goroutine 1410218 [running]:
github.com/microsoft/typescript-go/internal/ast.(*Node).Text(0x1406281ccb0?)
        /home/runner/work/tsgolint/tsgolint/typescript-go/internal/ast/ast.go:338 +0x418
github.com/typescript-eslint/tsgolint/internal/rules/no_deprecated.init.func1.7(0x140833eff28)
        /home/runner/work/tsgolint/tsgolint/internal/rules/no_deprecated/no_deprecated.go:408 +0x178
github.com/typescript-eslint/tsgolint/internal/rules/no_deprecated.init.func1.10(0x140833eff28)
        /home/runner/work/tsgolint/tsgolint/internal/rules/no_deprecated/no_deprecated.go:594 +0x298
github.com/typescript-eslint/tsgolint/internal/rules/no_deprecated.init.func1.12(0x10375fa60?)
        /home/runner/work/tsgolint/tsgolint/internal/rules/no_deprecated/no_deprecated.go:752 +0xf4
github.com/typescript-eslint/tsgolint/internal/linter.RunLinterOnProgram.func1.6(0x4f, 0x140833eff28)
        /home/runner/work/tsgolint/tsgolint/internal/linter/linter.go:253 +0x6c
github.com/typescript-eslint/tsgolint/internal/linter.RunLinterOnProgram.func1.8(0x140833eff28)
        /home/runner/work/tsgolint/tsgolint/internal/linter/linter.go:300 +0x48
github.com/microsoft/typescript-go/internal/ast.visit(...)
        /home/runner/work/tsgolint/tsgolint/typescript-go/internal/ast/ast.go:21
github.com/microsoft/typescript-go/internal/ast.(*BindingElement).ForEachChild(0x140833dc4d0, 0x14069346aa0)
        /home/runner/work/tsgolint/tsgolint/typescript-go/internal/ast/ast.go:3995 +0xb0
github.com/microsoft/typescript-go/internal/ast.(*Node).ForEachChild(0x1407af200d1?, 0x140833dc4d0?)
        /home/runner/work/tsgolint/tsgolint/typescript-go/internal/ast/ast.go:242 +0x24
github.com/typescript-eslint/tsgolint/internal/linter.RunLinterOnProgram.func1.8(0x140833dc4d0)
        /home/runner/work/tsgolint/tsgolint/internal/linter/linter.go:315 +0x134
github.com/microsoft/typescript-go/internal/ast.visitNodes(...)
        /home/runner/work/tsgolint/tsgolint/typescript-go/internal/ast/ast.go:28
github.com/microsoft/typescript-go/internal/ast.visitNodeList(...)
        /home/runner/work/tsgolint/tsgolint/typescript-go/internal/ast/ast.go:37
github.com/microsoft/typescript-go/internal/ast.(*BindingPattern).ForEachChild(0x18?, 0x14069346aa0)
        /home/runner/work/tsgolint/tsgolint/typescript-go/internal/ast/ast.go:3857 +0x74
github.com/microsoft/typescript-go/internal/ast.(*Node).ForEachChild(0xcf?, 0x14068ca2740?)
        /home/runner/work/tsgolint/tsgolint/typescript-go/internal/ast/ast.go:242 +0x24
github.com/typescript-eslint/tsgolint/internal/linter.RunLinterOnProgram.func1.8(0x14068ca2740)
        /home/runner/work/tsgolint/tsgolint/internal/linter/linter.go:315 +0x134
github.com/microsoft/typescript-go/internal/ast.visit(...)
        /home/runner/work/tsgolint/tsgolint/typescript-go/internal/ast/ast.go:21
github.com/microsoft/typescript-go/internal/ast.(*VariableDeclaration).ForEachChild(0x140497a4fd8, 0x14069346aa0)
        /home/runner/work/tsgolint/tsgolint/typescript-go/internal/ast/ast.go:3761 +0x40
github.com/microsoft/typescript-go/internal/ast.(*Node).ForEachChild(0x14063d50105?, 0x140497a4fd8?)
        /home/runner/work/tsgolint/tsgolint/typescript-go/internal/ast/ast.go:242 +0x24
github.com/typescript-eslint/tsgolint/internal/linter.RunLinterOnProgram.func1.8(0x140497a4fd8)
        /home/runner/work/tsgolint/tsgolint/internal/linter/linter.go:315 +0x134
github.com/microsoft/typescript-go/internal/ast.visitNodes(...)
        /home/runner/work/tsgolint/tsgolint/typescript-go/internal/ast/ast.go:28
github.com/microsoft/typescript-go/internal/ast.visitNodeList(...)
        /home/runner/work/tsgolint/tsgolint/typescript-go/internal/ast/ast.go:37
github.com/microsoft/typescript-go/internal/ast.(*VariableDeclarationList).ForEachChild(0x0?, 0x14069346aa0)
        /home/runner/work/tsgolint/tsgolint/typescript-go/internal/ast/ast.go:3811 +0x74
github.com/microsoft/typescript-go/internal/ast.(*Node).ForEachChild(0x1406a440106?, 0x1407af2fbc0?)
        /home/runner/work/tsgolint/tsgolint/typescript-go/internal/ast/ast.go:242 +0x24
github.com/typescript-eslint/tsgolint/internal/linter.RunLinterOnProgram.func1.8(0x1407af2fbc0)
        /home/runner/work/tsgolint/tsgolint/internal/linter/linter.go:315 +0x134
github.com/microsoft/typescript-go/internal/ast.visit(...)
        /home/runner/work/tsgolint/tsgolint/typescript-go/internal/ast/ast.go:21
github.com/microsoft/typescript-go/internal/ast.(*VariableStatement).ForEachChild(0x1406a44d8c8?, 0x103253e34?)
        /home/runner/work/tsgolint/tsgolint/typescript-go/internal/ast/ast.go:3707 +0x68
github.com/microsoft/typescript-go/internal/ast.(*Node).ForEachChild(0x14058cf00f4?, 0x140795e2230?)
        /home/runner/work/tsgolint/tsgolint/typescript-go/internal/ast/ast.go:242 +0x24
github.com/typescript-eslint/tsgolint/internal/linter.RunLinterOnProgram.func1.8(0x140795e2230)
        /home/runner/work/tsgolint/tsgolint/internal/linter/linter.go:315 +0x134
github.com/microsoft/typescript-go/internal/ast.visitNodes(...)
        /home/runner/work/tsgolint/tsgolint/typescript-go/internal/ast/ast.go:28
github.com/microsoft/typescript-go/internal/ast.visitNodeList(...)
        /home/runner/work/tsgolint/tsgolint/typescript-go/internal/ast/ast.go:37
github.com/microsoft/typescript-go/internal/ast.(*Block).ForEachChild(0x140813d8bd0?, 0x14069346aa0)
        /home/runner/work/tsgolint/tsgolint/typescript-go/internal/ast/ast.go:3664 +0x74
github.com/microsoft/typescript-go/internal/ast.(*Node).ForEachChild(0x1406a4400f2?, 0x14019c25a40?)
        /home/runner/work/tsgolint/tsgolint/typescript-go/internal/ast/ast.go:242 +0x24
github.com/typescript-eslint/tsgolint/internal/linter.RunLinterOnProgram.func1.8(0x14019c25a40)
        /home/runner/work/tsgolint/tsgolint/internal/linter/linter.go:315 +0x134
github.com/microsoft/typescript-go/internal/ast.visit(...)
        /home/runner/work/tsgolint/tsgolint/typescript-go/internal/ast/ast.go:21
github.com/microsoft/typescript-go/internal/ast.(*ArrowFunction).ForEachChild(0x140833e9c20, 0x14069346aa0)
        /home/runner/work/tsgolint/tsgolint/typescript-go/internal/ast/ast.go:6446 +0x168
github.com/microsoft/typescript-go/internal/ast.(*Node).ForEachChild(0xdc?, 0x140833e9c20?)
        /home/runner/work/tsgolint/tsgolint/typescript-go/internal/ast/ast.go:242 +0x24
github.com/typescript-eslint/tsgolint/internal/linter.RunLinterOnProgram.func1.8(0x140833e9c20)
        /home/runner/work/tsgolint/tsgolint/internal/linter/linter.go:315 +0x134
github.com/microsoft/typescript-go/internal/ast.visit(...)
        /home/runner/work/tsgolint/tsgolint/typescript-go/internal/ast/ast.go:21
github.com/microsoft/typescript-go/internal/ast.(*VariableDeclaration).ForEachChild(0x140706fc388, 0x14069346aa0)
        /home/runner/work/tsgolint/tsgolint/typescript-go/internal/ast/ast.go:3761 +0xe4
github.com/microsoft/typescript-go/internal/ast.(*Node).ForEachChild(0x14063d50105?, 0x140706fc388?)
        /home/runner/work/tsgolint/tsgolint/typescript-go/internal/ast/ast.go:242 +0x24
github.com/typescript-eslint/tsgolint/internal/linter.RunLinterOnProgram.func1.8(0x140706fc388)
        /home/runner/work/tsgolint/tsgolint/internal/linter/linter.go:315 +0x134
github.com/microsoft/typescript-go/internal/ast.visitNodes(...)
        /home/runner/work/tsgolint/tsgolint/typescript-go/internal/ast/ast.go:28
github.com/microsoft/typescript-go/internal/ast.visitNodeList(...)
        /home/runner/work/tsgolint/tsgolint/typescript-go/internal/ast/ast.go:37
github.com/microsoft/typescript-go/internal/ast.(*VariableDeclarationList).ForEachChild(0x0?, 0x14069346aa0)
        /home/runner/work/tsgolint/tsgolint/typescript-go/internal/ast/ast.go:3811 +0x74
github.com/microsoft/typescript-go/internal/ast.(*Node).ForEachChild(0x1404b300106?, 0x140708be200?)
        /home/runner/work/tsgolint/tsgolint/typescript-go/internal/ast/ast.go:242 +0x24
github.com/typescript-eslint/tsgolint/internal/linter.RunLinterOnProgram.func1.8(0x140708be200)
        /home/runner/work/tsgolint/tsgolint/internal/linter/linter.go:315 +0x134
github.com/microsoft/typescript-go/internal/ast.visit(...)
        /home/runner/work/tsgolint/tsgolint/typescript-go/internal/ast/ast.go:21
github.com/microsoft/typescript-go/internal/ast.(*VariableStatement).ForEachChild(0x14058cf7bf8?, 0x103253e34?)
        /home/runner/work/tsgolint/tsgolint/typescript-go/internal/ast/ast.go:3707 +0x68
github.com/microsoft/typescript-go/internal/ast.(*Node).ForEachChild(0x1404b3000f4?, 0x1406ff53b88?)
        /home/runner/work/tsgolint/tsgolint/typescript-go/internal/ast/ast.go:242 +0x24
github.com/typescript-eslint/tsgolint/internal/linter.RunLinterOnProgram.func1.8(0x1406ff53b88)
        /home/runner/work/tsgolint/tsgolint/internal/linter/linter.go:315 +0x134
github.com/microsoft/typescript-go/internal/ast.visitNodes(...)
        /home/runner/work/tsgolint/tsgolint/typescript-go/internal/ast/ast.go:28
github.com/microsoft/typescript-go/internal/ast.visitNodeList(...)
        /home/runner/work/tsgolint/tsgolint/typescript-go/internal/ast/ast.go:37
github.com/microsoft/typescript-go/internal/ast.(*Block).ForEachChild(0x14082d41b48?, 0x14069346aa0)
        /home/runner/work/tsgolint/tsgolint/typescript-go/internal/ast/ast.go:3664 +0x74
github.com/microsoft/typescript-go/internal/ast.(*Node).ForEachChild(0x1404b3000f2?, 0x14019c25aa0?)
        /home/runner/work/tsgolint/tsgolint/typescript-go/internal/ast/ast.go:242 +0x24
github.com/typescript-eslint/tsgolint/internal/linter.RunLinterOnProgram.func1.8(0x14019c25aa0)
        /home/runner/work/tsgolint/tsgolint/internal/linter/linter.go:315 +0x134
github.com/microsoft/typescript-go/internal/ast.visit(...)
        /home/runner/work/tsgolint/tsgolint/typescript-go/internal/ast/ast.go:21
github.com/microsoft/typescript-go/internal/ast.(*FunctionDeclaration).ForEachChild(0x1406d36c580, 0x14069346aa0)
        /home/runner/work/tsgolint/tsgolint/typescript-go/internal/ast/ast.go:4089 +0x1a0
github.com/microsoft/typescript-go/internal/ast.(*Node).ForEachChild(0x400000107?, 0x1406d36c580?)
        /home/runner/work/tsgolint/tsgolint/typescript-go/internal/ast/ast.go:242 +0x24
github.com/typescript-eslint/tsgolint/internal/linter.RunLinterOnProgram.func1.8(0x1406d36c580)
        /home/runner/work/tsgolint/tsgolint/internal/linter/linter.go:315 +0x134
github.com/microsoft/typescript-go/internal/ast.visitNodes(...)
        /home/runner/work/tsgolint/tsgolint/typescript-go/internal/ast/ast.go:28
github.com/microsoft/typescript-go/internal/ast.visitNodeList(...)
        /home/runner/work/tsgolint/tsgolint/typescript-go/internal/ast/ast.go:37
github.com/microsoft/typescript-go/internal/ast.(*SourceFile).ForEachChild(0x1405aaa5208, 0x14069346aa0)
        /home/runner/work/tsgolint/tsgolint/typescript-go/internal/ast/ast.go:10845 +0xa4
github.com/microsoft/typescript-go/internal/ast.(*Node).ForEachChild(0x10375fa60?, 0x1406933b2c0?)
        /home/runner/work/tsgolint/tsgolint/typescript-go/internal/ast/ast.go:242 +0x24
github.com/typescript-eslint/tsgolint/internal/linter.RunLinterOnProgram.func1()
        /home/runner/work/tsgolint/tsgolint/internal/linter/linter.go:323 +0xd0
github.com/microsoft/typescript-go/internal/core.(*parallelWorkGroup).Queue.func1()
        /home/runner/work/tsgolint/tsgolint/typescript-go/internal/core/workgroup.go:42 +0x50
created by github.com/microsoft/typescript-go/internal/core.(*parallelWorkGroup).Queue in goroutine 1
        /home/runner/work/tsgolint/tsgolint/typescript-go/internal/core/workgroup.go:40 +0x7c
Error running tsgolint: "exit status: exit status: 2"⏎  
```

</details>

**Versions:**
macOS: 15.5
node: 23.11.0
oxlint: 1.26.0
oxlint-tsgolint: 0.5.0

--- a/internal/rules/no_deprecated/no_deprecated.go
+++ b/internal/rules/no_deprecated/no_deprecated.go
@@ -314,6 +314,9 @@ var NoDeprecatedRule = rule.Rule{
 						// Get the property name for this binding element
 						propertyName := ""
 						if bindingElem.PropertyName != nil {
+							if bindingElem.PropertyName.Kind == ast.KindComputedPropertyName {
+								return nil
+							}
 							propertyName = bindingElem.PropertyName.Text()
 						} else if bindingElem.Name() != nil {
 							name := bindingElem.Name()
@@ -405,6 +408,9 @@ var NoDeprecatedRule = rule.Rule{
 							// Get the property name being destructured
 							propertyName := node.Text()
 							if bindingElem.PropertyName != nil {
+								if bindingElem.PropertyName.Kind == ast.KindComputedPropertyName {
+									return false, ""
+								}
 								propertyName = bindingElem.PropertyName.Text()
 							}
 

--- a/internal/rules/no_deprecated/no_deprecated_test.go
+++ b/internal/rules/no_deprecated/no_deprecated_test.go
@@ -487,9 +487,18 @@ exists('/foo');
         b: 'string',
       };
 
-      const key = null;
-      const c = a[key as any];
-    `},
+		const key = null;
+		const c = a[key as any];
+	`},
+		{Code: `
+		interface MyInterface {
+			/** @deprecated */
+			prop: string;
+		}
+		const obj: MyInterface = { prop: 'value' };
+		const key = 'prop';
+		const { [key]: value } = obj;
+	`},
 	}, []rule_tester.InvalidTestCase{
 		{
 			Tsx: true,

